cli/config/configfile/file.go

```diff
@@ -254,10 +255,13 @@ func decodeAuth(authStr string) (string, string, error) {
 // GetCredentialsStore returns a new credentials store from the settings in the
 // configuration file
 func (configFile *ConfigFile) GetCredentialsStore(registryHostname string) credentials.Store {
+	var credsStore credentials.Store
 	if helper := getConfiguredCredentialStore(configFile, registryHostname); helper != "" {
-		return newNativeStore(configFile, helper)
+		credsStore = newNativeStore(configFile, helper)
+	} else {
+		credsStore = credentials.NewFileStore(configFile)
 	}
-	return credentials.NewFileStore(configFile)
+	return credentials.NewOAuthStore(credsStore, manager.NewManager())
 }
```

Comment:
This is not something new, but I think your PR makes it more apparent, and ... maybe I'm rambling; I need to have a closer look; I'm somewhat starting to consider if constructing these stores should happen here, or if they should be constructed when initialising the configfile.

Basically; I need to look where this code is used; it may be imported elsewhere, which can be in code that's "non-interactive" (say k8s); for those, getting credentials through the OAuth flow isn't an option.

So I'm thinking if whether or not to use the interactive flow is something that must be set when constructing all of this 🤔

Reply:
>  for those, getting credentials through the OAuth flow isn't an option

True, but as you said, it's not different/new here. With the "old" flow, if there are no credentials, we also can't prompt the user to input them if running non-interactive. And if the user does have "oauth" credentials stored, even in a non-interactive environment, we want the oauth codepath to be executed so that the CLI can use the refresh token to fetch a fresh access token before returning them to whoever is calling it, right?

Basically, what setting this when initializing let us do/decide that we can't already?